core/thread.c

```diff
@@ -124,6 +124,19 @@ void thread_add_to_list(list_node_t *list, thread_t *thread)
     list->next = new_node;
 }
 
+void thread_remove_from_list(list_node_t *list, thread_t *thread)
```

Comment:
Alternatively: why not add a `list_remove()` implementation? Seems a lot cleaner.